include/dlaf/matrix/panel.h

```diff
@@ -0,0 +1,362 @@
+
+    return index;
+  }
+
+  /// Given a matrix index, check if the corresponding tile in the panel is external or not
+  bool is_external(const LocalTileIndex idx_matrix) const noexcept {
+    return external_[linear_index(idx_matrix)].valid();
+  }
+
+  ///> Parent matrix which this panel is related to
+  Distribution dist_matrix_;
+
+  ///> It represents from where it is necessary to allocate memory
+  SizeType bias_;
+  ///> It represents from where the panel gives access to tiles
+  SizeType offset_;
+  ///> Interanlly store the range of tiles accessible (changes according to @p offset_)
+  iter2d_t range_;
+
+  ///> Container for references to external tiles
+  common::internal::vector<hpx::shared_future<ConstTileT>> external_;
+  ///> Keep track of usage status of internal tiles (accessed or not)
+  std::set<SizeType> internal_;
+};
+
+template <Coord axis, class T, Device device>
+struct Panel : public Panel<axis, const T, device> {
+  using TileT = Tile<T, device>;
+  using ConstTileT = Tile<const T, device>;
+
+  explicit Panel(matrix::Distribution distribution, LocalTileSize start = {0, 0})
+      : Panel<axis, const T, device>(std::move(distribution), std::move(start)) {}
+
+  /// Access tile at specified index in readwrite mode
+  ///
+  /// It is possible to access just internal tiles in RW mode.
+  ///
+  /// @pre index must point to a tile which is internally managed by the panel
+  hpx::future<TileT> operator()(const LocalTileIndex& index) {
+    DLAF_ASSERT(!is_external(index), "read-only access on external tiles", index);
+
+    BaseT::internal_.insert(BaseT::linear_index(index));
+    return BaseT::operator()(BaseT::full_index(index));
+  }
+
+protected:
+  using BaseT = Panel<axis, const T, device>;
+  using BaseT::is_external;
+};
+
+namespace internal {
+
+// helper function that identifies the owner of a transposed coordinate,
+// it returns both the component of the rank in the transposed dimension and
+// its global cross coordinate (i.e. row == col in the global frame of reference)
+template <Coord dst_coord>
+std::pair<SizeType, comm::IndexT_MPI> transposed_owner(const Distribution& dist,
+                                                       const LocalTileIndex idx) {
+  const auto idx_cross = dist.template globalTileFromLocalTile<dst_coord>(idx.get(dst_coord));
+  const auto rank_owner = dist.template rankGlobalTile<orthogonal(dst_coord)>(idx_cross);
+  return std::make_pair(idx_cross, rank_owner);
+}
+
+}
+
+/// Broadcast a panel in the direction orthogonal to its axis
+template <class T, Device device, Coord axis>
+void broadcast(hpx::execution::parallel_executor ex, comm::IndexT_MPI rank_root,
+               Panel<axis, T, device>& ws, common::Pipeline<comm::CommunicatorGrid>& serial_comm) {
+  using namespace comm::sync::broadcast;
+  using hpx::dataflow;
+
+  constexpr auto comm_dir = orthogonal(axis);
+
+  // TODO
+  // if (grid_size.get(component(comm_dir)) < 1)
+  //  return;
```

Comment:
This is an open topic (it applies also to the other `broadcast` function): there is no way of knowing the size of the communicator before scheduling the tasks, unless we explicitly decide to pass it as argument.

If not checked, we may schedule almost "empty" tasks because the check will be carried out in nested calls once the task is in execution (e.g. in the MPI send/recv function).

Reply:
@rasolca instead of passing as argument the communicator size, we could opt for getting the information from the `parentDistribution` as alternative.

What do you think it is better?